docs/learn/learn-xcvm.md

```diff
@@ -8,28 +8,108 @@ slug: ../learn-xcvm
+chain may treat incoming messages via
+[Barriers](https://github.com/paritytech/polkadot/blob/26b0c4f6273190f4538b24939a56b6a0b51a344c/xcm/xcm-executor/src/config.rs#L52),
+calculate weight for a message via the
+[Weigher](https://github.com/paritytech/polkadot/blob/26b0c4f6273190f4538b24939a56b6a0b51a344c/xcm/xcm-executor/src/config.rs#L55),
+how much weight to purchase via the
+[Trader](https://github.com/paritytech/polkadot/blob/26b0c4f6273190f4538b24939a56b6a0b51a344c/xcm/xcm-executor/src/config.rs#L58),
+[configure fees](https://github.com/paritytech/polkadot/blob/26b0c4f6273190f4538b24939a56b6a0b51a344c/xcm/xcm-executor/src/config.rs#L89),
+how to
+[convert origins](https://github.com/paritytech/polkadot/blob/26b0c4f6273190f4538b24939a56b6a0b51a344c/xcm/xcm-executor/src/config.rs#L40),
+and more.
+
+## Cross Consensus Message (XCM) Anatomy & Flow
+
+An XCM is made up of a list of instructions that are executed in order. There are four different
+kinds of XCM instructions:
 
-One example of such an instruction would be `TransferAsset` which is used to transfer an asset to
-some other address on the remote system. It needs to be told which asset(s) to transfer and to
-whom/where the asset is to be transferred.
+1. **Instruction** - Results in a state change in the local consensus system or some state change.
+2. **Trusted Indication** - Tells the XCVM, or the Executor, that some action has been done before
+   already - meaning, this action is now trusted and can be acted on, i.e., in a teleport scenario.
+3. **Information** - Provides additional information about a particular origin, usually the result
+   of a query, i.e., a `QueryResponse` instruction.
+4. **System Notification** - Typically used in the context of when an HRMP channel is being opened,
+   closed, or accepted.
 
+Typically, an XCM takes the following path:
+
+1.  The XCM Executor, and XCVM in general, is very much analogous to a CPU. It has various
+    registers, which instructions usually modify as needed.
+2.  The resulting values from these registers are used later in the transfer. An example of a common
+    register used in many instructions is the `Holding` register, which places assets in a holding
+    until fulfilled on the destination chain.
+3.  The resulting instructions modify the Holding register as needed and execute extrinsics on the
+    destination chain to perform the state change. The Executor is able to convert incoming
+    instructions into FRAME-compatible origins, which enables the actual state changes to take
+    place.
+
+### Example Register: The Holding Register
+
+There are many instructions that depend on the Holding register. The Holding register is an XCM
+register that provides a place for any assets that are in an intermediary state to be held until
+they are withdrawn to a beneficiary. They require an instruction to place assets within and another
```

Comment:
I think using the word "withdraw" for removing assets from the holding register might be confusing since the instructions usually call it `Deposit` and `Withdraw` is the opposite operation

Reply:
Ah, so to be clear: to _deposit_ means to take from the holding register, while withdraw means to withdraw assets _to_ the holding register?

Reply:
Yeah, at least that's the logic with the instructions `DepositAsset` and `WithdrawAsset`. They act on accounts, where they `deposit` or `withdraw` but as a side effect end up taking from or putting in the holding register respectively.